Enhance Network Connection Checker Implementation

M-Abdelnabi opened the pull request.

M-Abdelnabi committed:
Feat: Add network permissions to home module

The `repository/home` module now requests `INTERNET` and `ACCESS_NETWORK_STATE` permissions in its `AndroidManifest.xml`. This enables the module to perform network operations.

M-Abdelnabi committed:
Refactor: Relocate AndroidManifest.xml for search module

The `AndroidManifest.xml` file for the `repository/search` module has been moved from `repository/search/src/main/java/com/repository/search/` to `repository/search/src/main/`.

This change aligns with standard Android project structure conventions, where the manifest file is typically located directly within the `src/main` directory of a module.

M-Abdelnabi committed:
call startChecker() inside init{} of the class and rename the class to HomeNetworkConnectionChecker.kt

M-Abdelnabi committed:
call startChecker() inside init{} of the class and rename the class to MovieNetworkConnectionChecker.kt

M-Abdelnabi committed:
call startChecker() inside init{} of the class and rename the class to SearchNetworkConnectionChecker.kt

M-Abdelnabi committed:
call startChecker() inside init{} of the class and rename the class to TvNetworkConnectionChecker.kt

M-Abdelnabi committed:
Refactor: Rename `NetworkConnectionChecker` to `TvNetworkConnectionChecker`

Renamed `NetworkConnectionChecker` to `TvNetworkConnectionChecker` in `TvShowRepositoryImpl.kt` and its corresponding test file `TvShowRepositoryImplTest.kt`. This change improves clarity and specificity regarding the component's role within the TV show domain.

M-Abdelnabi committed:
Refactor: Rename `NetworkConnectionChecker` to `SearchNetworkConnectionChecker`

Renamed `NetworkConnectionChecker` to `SearchNetworkConnectionChecker` in `SearchMediaRepositoryImpl.kt` and `SearchMediaRepositoryImplTest.kt` for better clarity and to avoid potential naming conflicts with other modules.

M-Abdelnabi committed:
Refactor: Rename `NetworkConnectionChecker` to `MovieNetworkConnectionChecker`

Renamed `NetworkConnectionChecker` to `MovieNetworkConnectionChecker` in `MovieRepositoryImpl.kt` and `MovieRepositoryImplTest.kt` for better clarity and specificity within the movie module.

M-Abdelnabi committed:
Refactor: Rename `NetworkConnectionChecker` to `HomeNetworkConnectionChecker`

The `NetworkConnectionChecker` class and its instances have been renamed to `HomeNetworkConnectionChecker` in `MediaRepositoryImpl.kt` and `MediaRepositoryImplTest.kt`. This change improves clarity and specificity regarding the checker's context within the home module.

M-Abdelnabi committed:
Refactor: Rename `NetworkConnectionChecker` to `SearchNetworkConnectionChecker`

Renamed `NetworkConnectionChecker` to `SearchNetworkConnectionChecker` in `CategoriesRepositoryImpl` and its corresponding test class `CategoriesRepositoryImplTest`.

This change clarifies the specific context of the network connection checker, associating it directly with search-related functionalities.

M-Abdelnabi committed:
Refactor: Consolidate NetworkConnectionChecker classes

The four separate `NetworkConnectionChecker` classes within the `repository` submodules (`home`, `movie`, `search`, and `util`) have been consolidated into their respective utility packages.

This change simplifies the dependency injection setup in `NetworkModule.kt` by directly referencing these more specifically named checker classes:
- `com.repository.search.util.NetworkConnectionChecker` is now `com.repository.search.util.SearchNetworkConnectionChecker`
- `com.repository.util.NetworkConnectionChecker` is now `com.repository.util.TvNetworkConnectionChecker`
- `com.repository.movie.util.NetworkConnectionChecker` is now `com.repository.movie.util.MovieNetworkConnectionChecker`
- `com.repository.home.util.NetworkConnectionChecker` is now `com.repository.home.util.HomeNetworkConnectionChecker`

M-Abdelnabi committed:
Refactor: Simplify network checker initialization

The initialization of network connection checkers in `AflamiApplication.kt` has been simplified. Instead of obtaining instances from Koin and calling `startChecker()`, they are now directly instantiated.

M-Abdelnabi committed:
Test: Reset mocks before each LoginViewModelTest

The `clearAllMocks()` function from MockK has been added to the `@BeforeEach` setup method in `LoginViewModelTest.kt`.

This ensures that all mocks are reset to their initial state before each test execution, preventing potential interference between tests and ensuring test isolation.

Joseph-Sameh-0 commented on app/src/main/java/com/paris_2/aflami/AflamiApplication.kt:
```diff
@@ -55,13 +57,9 @@ class AflamiApplication : Application() {
-        networkChecker2.startChecker()
-        networkChecker3.startChecker()
-        networkChecker4.startChecker()
+        TvNetworkConnectionChecker(this)
+        SearchNetworkConnectionChecker(this)
+        MovieNetworkConnectionChecker(this)
+        HomeNetworkConnectionChecker(this)
```
you don't need this part remove it

M-Abdelnabi committed:
Refactor: Simplify network checker initialization

The initialization of network connection checkers in `AflamiApplication.kt` has been simplified. Instead of obtaining instances from Koin and calling `startChecker()`, they are now directly instantiated.

M-Abdelnabi committed:
Merge branch 'develop' into refactor/network-checker

M-Abdelnabi committed:
Update version name to 0.3.5 and release notes

The `versionName` in `app/release-info.txt` has been updated from `0.3.4` to `0.3.5`.
The release notes have been changed from "update exception handing in MoviesCategoriesRepository" to "enhance network checker implementation".

M-Abdelnabi committed:
Refactor: Update NetworkConnectionChecker type

The type of `networkConnectionChecker` in `MoviesCategoriesRepositoryImpl.kt` has been changed from `NetworkConnectionChecker` to `HomeNetworkConnectionChecker`.

M-Abdelnabi committed:
Refactor: Update NetworkConnectionChecker type

The type of `networkChecker` in `MoviesCategoriesRepositoryImplTest.kt` has been changed from `NetworkConnectionChecker` to `HomeNetworkConnectionChecker`.

M-Abdelnabi committed:
Merge branch 'develop' into refactor/network-checker

# Conflicts:
#	app/release-info.txt

M-Abdelnabi committed:
Update app version and release notes

The `versionName` in `app/release-info.txt` has been updated from `0.3.5` to `0.3.6`.
The release notes have been changed from "- refactor play video for episode" to "- enhance network checker implementation".

M-Abdelnabi committed:
Update app version and release notes

The `versionName` in `app/release-info.txt` has been updated from `0.3.5` to `0.3.6`.
The release notes have been changed from "- refactor play video for episode" to "- enhance network checker implementation".

M-Abdelnabi committed:
Merge branch 'develop' into refactor/network-checker

# Conflicts:
#	app/release-info.txt
#	app/src/main/java/com/paris_2/aflami/AflamiApplication.kt
#	app/src/main/java/com/paris_2/aflami/di/NetworkModule.kt
#	feature/authentication/authenticationUi/src/test/java/com/feature/authentication/authenticationUi/screen/login/LoginViewModelTest.kt
#	repository/search/src/main/java/com/repository/search/repository/CategoriesRepositoryImpl.kt
#	repository/search/src/test/kotlin/com/repository/search/repository/CategoriesRepositoryImplTest.kt
#	repository/tvShows/src/test/java/com/repository/tvshow/repository/TvShowRepositoryImplTest.kt

M-Abdelnabi committed:
refactor: replace NetworkConnectionChecker with SearchNetworkConnectionChecker in CategoriesRepositoryImpl

M-Abdelnabi committed:
refactor: update CategoriesRepositoryImplTest to use SearchNetworkConnectionChecker

M-Abdelnabi committed:
merge develop into network-checker
